### PR TITLE
Compile C# byte[] to Buffer

### LINKED ISSUE #217: Contract compile issues for Helper.Reverse()
A contract compiler error for Helper.Reverse(this byte[] source),like:
image

### LINKED ISSUE #225: Compile C# byte[] to Buffer

### PR DESCRIPTION
**Pull request by ShawnYun**
close #217 #218 #225

1. Compile C# byte[] to Buffer
2. Add Helper UnitTest.

### CODE REVIEW

File: src/Neo.SmartContract.Framework/Helper.cs [hunk 1]
```diff
@@ -194,18 +194,20 @@ public static byte ToByte(this int source)
...
+        //    for (int k = 0; k < source.Length / 2; k++)
+        //    {
+        //        int m = source.Length - k - 1;
+        //        byte bg = source[k]; // must store on variable before next assignment
+        //        byte ed = source[m]; // must store on variable before next assignment
+        //        source[k] = ed;
+        //        source[m] = bg;
+        //    }
+        //    return source;
+        //}
```
Comment (erikzhang) [inline src/Neo.SmartContract.Framework/Helper.cs:210]: The old code should be deleted.

**Comment by ProDog** [comment]
I tested it passed.

Contract:

image

Invoke result:

image

File: src/Neo.SmartContract.Framework/Helper.cs [hunk 2]
```diff
@@ -27,19 +25,19 @@ public static class Helper
...
-        [OpCode(OpCode.CONVERT, StackItemType_ByteArray)]
+        [OpCode(OpCode.CONVERT, StackItemType_Buffer)]
         public extern static byte[] ToByteArray(this sbyte source);
 
         /// <summary>
         /// Converts sbyte[] to byte[].
         /// </summary>
-        [OpCode(OpCode.CONVERT, StackItemType_ByteArray)]
+        [OpCode(OpCode.CONVERT, StackItemType_Buffer)]
         public extern static byte[] ToByteArray(this sbyte[] source);
```
Comment (erikzhang) [inline src/Neo.SmartContract.Framework/Helper.cs:35]: What type does `sbyte[]` correspond to in neo-vm?

Comment (ShawnYun) [reply src/Neo.SmartContract.Framework/Helper.cs:35]: @erikzhang It is also a byte[] with complement.

Comment (erikzhang) [reply src/Neo.SmartContract.Framework/Helper.cs:35]: Then is `CONVERT` necessary?

Comment (ShawnYun) [reply src/Neo.SmartContract.Framework/Helper.cs:35]: @erikzhang  Yes, it should do noting.

**Comment by shargon** [comment]
Could you split the rename and the compilation change in two PR, it will be easier for review

**Comment by shargon** [comment]
My fault, you can revert the split 👼

**Comment by ShawnYun** [comment]
@erikzhang @shargon  Has been modified.

**Comment by ShawnYun** [comment]
Just rename "reverseArray" to "wantResult".

**Pull request merged**